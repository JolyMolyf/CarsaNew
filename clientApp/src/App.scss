@import "./sass/variables.scss";

*,
*::before,
*::after {
  margin: 0;
  padding: 0;
}

html {
  font-size: 62.5%; // from now 1rem === 10px
}

body {
  font-size: 1.6rem; // we set default font size value to the 16px as it was
  background-color: $backgroundColor;
  box-sizing: border-box;
}
