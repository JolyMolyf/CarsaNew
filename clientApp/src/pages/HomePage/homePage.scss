@import '../../sass//variables.scss';

.home{
  position: absolute;
  background-color: #fff;
  min-height: 100vh;
  &-decorations{
    width: 22px;
    position: absolute;
    left: 40px;
    top: 35vh;
    display: flex;
    justify-content: space-between;
    flex-direction: row;
    &-item{
      &:nth-child(1){
        margin-top: 100px;
      }

      width: 2px;
      height: 30vh;
      background-color: #000;

    }

    &-horizontal{
      transform: rotate(90deg);
      height: 5%;
      top: 0; 
      left: 90%;

      &:nth-child(2){
        top: 75vh !important;
        left: 30% !important;
      }
    }

    &:nth-child(2){
      top: 10vh;
      left: 96%;

    }
  }


  &-section{
    min-height: 80vh;
    width: 100%;
    position: relative;
    &-first{
      margin-top: 100px;
      display: flex;
      justify-content: center;
      flex-wrap: wrap;
      &-fadingText{
        font-family: Montserrat;
        font-style: normal;
        font-weight: 900;
        font-size: 288px;
        line-height: 351px;
        color: #FFB100;
        background: -webkit-linear-gradient(top,#ffb300a8 0%, #fff);
        -webkit-background-clip: text;
        -webkit-text-fill-color: transparent;
      }
      img{
        margin-top: -220px;
      }
    }

    &-marketing{
      display: flex;
      width: 50%;
      margin: 0 auto;
      justify-content: space-evenly;
      align-items: center;
      font-family: 'PT Sans Narrow', sans-serif;
     
      p{
        font-style: normal;
        font-weight: normal;
        font-size: 36px;
        line-height: 52px;
        font-weight: 700;
        text-transform: uppercase;
      }

      img{ 
        width: 178px;
        height: 30px;
        transform: scale(0.5);
       }
       &-btn{
         width: 150px;
         height: 150px;
         text-align: center;
         line-height: 150px;
         cursor: pointer;
         text-transform: uppercase;
         background:radial-gradient(50% 50% at 50% 50%, rgba(255, 220, 142, 0.84) 0%, rgba(219, 163, 35, 0) 100%);
       }
    }

    
  }
  &-second{
    height: 80vh;
    background-color: #ffedc454;
    position: relative;
    overflow: hidden;

    &-main{
      display: flex;
      justify-content: space-between;
      flex-direction: row;
      align-items: center;
      padding-top: 100px;
      padding-bottom: 100px;
      position: relative;
      width: 100%;
      height: 60vh;
      padding-left: 10%;
      &-body{
        font-family: 'PT Sans Narrow', sans-serif;
        font-family: 'Space Mono', monospace;
        &-pre{
          color: #FFD77F;
          font-style: normal;
          font-weight: 700;
          font-size: 24px;
          line-height: 35px;
        }

        &-header{
          font-style: normal;
          font-weight: normal;
          font-size: 36px;
          line-height: 52px;
        }
        &-section{
          margin-top: 20px;
          margin-bottom: 20px;
          font-family: 'PT Sans Narrow', sans-serif;
          font-family: 'Space Mono', monospace;
          &-header{
            font-style: normal;
            font-weight: normal;
            font-size: 21px;
            line-height: 30px;
            display: flex;
            align-items: center;
            letter-spacing: 0.205em;
          }

          &-body{

            font-style: normal;
            font-weight: normal;
            font-size: 13px;
            line-height: 19px;  
          }
        }

      }
      &-image{
        img{

        }
      }
    }
  }

  &-third{
    width: 100%;
    text-align: center;
    font-family: 'PT Sans Narrow', sans-serif;
    font-family: 'Space Mono', monospace;
    &-header{
      color: #FFD77F;
      font-style: normal;
      font-weight: 700;
      font-size: 48px;
      line-height: 70px;
    }  

    &-body{
      &-reviews{
        display: flex;  
        flex-direction: row;
        justify-content: space-between;
        margin: 0 auto;
        max-width: 1150px;

        &-item{
          width: 350px;
        }

      }
    }
  }
}
