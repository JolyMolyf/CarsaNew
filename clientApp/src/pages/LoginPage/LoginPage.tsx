import React from 'react'
import Header from '../../components/header/Header'
import styles from './loginPage.module.scss'
import LoginForm from '../../components/loginForm/LoginForm'

const LoginPage = () => {
	return (
		<div className={styles.wrapper}>
			<Header />
			<LoginForm />
		</div>
	)
}

export default LoginPage
