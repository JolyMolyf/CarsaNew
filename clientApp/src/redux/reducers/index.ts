import { combineReducers } from 'redux'
import { persistReducer  } from 'redux-persist'
import  storage  from 'redux-persist/lib/storage'
import appReducer from './appReducer'
import authReducer from './auth'
import carReducer from './carReducer'
import orderReducer from './orderReducer'
import paymentsReducer from "./paymentsReducer";

const persistConfig = {
  key: 'root',
  storage,
  version: 1,
  whitelist: ['auth']
}


const rootReducer = combineReducers({
  cars: carReducer,
  auth: authReducer,
  orders: orderReducer,
  payments: paymentsReducer,
  app: appReducer,
})

export default persistReducer(persistConfig, rootReducer)
