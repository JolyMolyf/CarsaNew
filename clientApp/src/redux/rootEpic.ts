import { combineEpics } from "redux-observable";
import { saveCarsEpic, getCarsEpic, getCarClient } from "./epics/carEpic";
import { saveUser } from "./epics/userEpic";
import { getOrderEpic, saveOrderEpic } from "./epics/orderEpic";
import { getPaymentsEpic, savePaymentsEpic } from "./epics/paymentsEpic";

export const rootEpic = combineEpics(
  saveCarsEpic, getCarsEpic, getCarClient, saveUser, getOrderEpic, saveOrderEpic, getPaymentsEpic, savePaymentsEpic
);

