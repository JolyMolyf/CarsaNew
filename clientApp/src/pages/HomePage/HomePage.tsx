import React from 'react'
import HomeHeader from '../../components/Home/Header/HomeHeader'
import './homePage.scss'
import brnadCar from '../../images/HomePage/brandCar.png'
import arrow from '../../images/HomePage/Arrow 2.png'
const HomePage = () => {
   

    return (
        <div className='home'>
            <div className='home-decorations'>
                <div className='home-decorations-item'></div>
                <div className='home-decorations-item'></div>
            </div>
            <div className='home-decorations'>
                <div className='home-decorations-item'></div>
                <div className='home-decorations-item'></div>
            </div>
            <HomeHeader></HomeHeader>
            <div className='home-section'>
                <div className='home-section-first'>
                    <div className='home-section-first-fadingText'>
                        CARSA
                    </div>
                    <img src={brnadCar}></img>
                </div>
                <div className='home-section-marketing'>
                    <p>Selection helper</p>
                    <img src={arrow}></img>
                    <div className='home-section-marketing-btn'>Drive Now</div>
                </div>
                  
            </div>
            
        
        </div>
    )
}

export default HomePage