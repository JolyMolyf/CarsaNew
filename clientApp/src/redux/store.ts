import {applyMiddleware, createStore} from "redux";
import {composeWithDevTools} from 'redux-devtools-extension';
import rootReducer from './reducers' 
import { createEpicMiddleware } from "redux-observable";
import { rootEpic } from './rootEpic';
import { persistStore } from 'redux-persist'




export  function configureStore() {
  const epicMiddleware = createEpicMiddleware(); 
    let store = createStore(rootReducer,  composeWithDevTools(
    applyMiddleware(epicMiddleware)
    // other store enhancers if any
  ))

  epicMiddleware.run(rootEpic);
  return store;
};

export type storeState = ReturnType<typeof store.getState>
export const getStore =  () => {
  return store.getState();
}

export const store =  configureStore();

export const persistor = persistStore(store);

