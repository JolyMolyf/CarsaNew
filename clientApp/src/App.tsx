import React from "react";
import logo from "./logo.svg";
import "./App.scss";
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Link,
    Redirect,
} from "react-router-dom";
import HomePage from "./pages/HomePage/HomePage";
import OfferedCars from "./pages/OfferedCars/OfferedCars";
import CarPage from "./pages/CarPage/CarPage";
import LoginPage from "./pages/LoginPage/LoginPage";
import RegistrationPage from "./pages/RegistrationPage/RegistrationPage";
import ProtectedRoute from "./components/shared/protectedRoute/ProtectedRoute";
import Dashboard from "./pages/DashBoard/DashBoard";
import OrderDashBoard from "./pages/DashBoard/OrderDashBoard/OrderDashBoard";
import PaymentDashBoard from "./pages/DashBoard/PaymentsDashBoard/PaymentsDashBoard";
import CarDashBoard from "./pages/DashBoard/CarDashBoard/CarDashBoard";
// import CreateOrder from "./pages/DashBoard/Order/CreateOrder/createOrder";
import CreateReport from "./pages/ReportPages/CreateReport/CreateReport";
import ReportMainPage from "./pages/ReportPages/ReportMainPage/ReportMainPage";
function App() {


    return (
        <div className="App">
            <Router>
                <Switch>
                    <Route exact path="/home" component={HomePage}/>
                    <Route exact path="/offers" component={OfferedCars}/>
                    <Route exact path="/login" component={LoginPage}/>
                    <Route exact path="/register" component={RegistrationPage}/>
                    <Route exact path="/car/:id" component={CarPage}/>
                    <ProtectedRoute exact path="/dashboard" component={Dashboard}/>
                    <ProtectedRoute exact path="/dashboard/orders" component={OrderDashBoard}/>
                    <ProtectedRoute exact path="/dashboard/cars" component={CarDashBoard}/>
                    <ProtectedRoute exact path="/dashboard/payments" component={PaymentDashBoard}/>
                    <ProtectedRoute exact path="/report" component={ReportMainPage}/>
                    <Route exact path='/report/:id' component={CreateReport}/>
                    {/* <Route exact path='/dashboard/createorder' component={CreateOrder}/> */}
                    <Route exact path="/" render={() => <Redirect to="/home"/>}/>
                </Switch>
            </Router>
        </div>
    );
}

export default App;
