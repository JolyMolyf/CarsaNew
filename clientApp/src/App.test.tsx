
import App from './App';
import renderer from 'react-test-renderer'
import reducer from './redux/reducers'
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import HomePage from './pages/HomePage/HomePage';
import LoginPage from './pages/LoginPage/LoginPage';
import {createMemoryHistory} from 'history'
import {Router} from 'react-router-dom'

const AppWrapper = () => {
  const store = createStore(reducer);

  return (
    <Provider store={store}> // Set context
      <App /> // Now App has access to context
    </Provider>
  )
}

const HomeWrapper = () => {
  const store = createStore(reducer);
  const history = createMemoryHistory()
  history.push('/some/bad/route')
  return (
    <Provider store={store}> // Set context
     <Router history={history}>
      <HomePage /> // Now App has access to context
      </Router>
    </Provider>
  )
}
const LogInWrapper = () => {
  const store = createStore(reducer);
  const history = createMemoryHistory()
  history.push('/some/bad/route')
  return (
    <Provider store={store}> // Set context
     <Router history={history}>
      <LoginPage /> // Now App has access to context
      </Router>
    </Provider>
  )
}


describe('components should render correct', () => {

  test('app snapshot test', () => {
    const tree = renderer.create(<AppWrapper/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('home page test', () => {
    const tree = renderer.create(<HomeWrapper/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  test('login page test', () => {
    const tree = renderer.create(<LogInWrapper/>).toJSON();
    expect(tree).toMatchSnapshot();
  })
})




