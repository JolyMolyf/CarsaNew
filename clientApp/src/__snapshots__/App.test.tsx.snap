// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components should render correct app snapshot test 1`] = `
Array [
  " // Set context",
  <div
    className="App"
  >
    <div
      className="home"
    >
      <div
        className="home-decorations"
      >
        <div
          className="home-decorations-item"
        />
        <div
          className="home-decorations-item"
        />
      </div>
      <div
        className="home-decorations"
      >
        <div
          className="home-decorations-item"
        />
        <div
          className="home-decorations-item"
        />
      </div>
      <div
        className="homeheader"
      >
        <div
          className="homeheader-wrapper"
        >
          <a
            href="/home"
            onClick={[Function]}
          >
            Home
          </a>
        </div>
        <div
          className="homeheader-wrapper"
        >
          <a
            href="/offers"
            onClick={[Function]}
          >
            Offers
          </a>
        </div>
        
        <div
          className="homeheader-logo"
        >
          <img
            src="logo.png"
          />
        </div>
        
        <div
          className="homeheader-wrapper"
        >
          <a
            href="/about"
            onClick={[Function]}
          >
            Contact
          </a>
        </div>
        <div
          className="homeheader-wrapper"
        >
          <a
            className="header-wrapper-link"
            href="/login"
            onClick={[Function]}
          >
            Log in
          </a>
        </div>
      </div>
      <div
        className="home-section"
      >
        <div
          className="home-section-first"
        >
          <div
            className="home-section-first-fadingText"
            data-aos="fade-up"
          >
            CARSA
          </div>
          <img
            data-aos="fade-down"
            src="brandCar.png"
          />
        </div>
        <div
          className="home-section-marketing"
        >
          <p>
            Selection helper
          </p>
          <img
            src="Arrow 2.png"
          />
          <div
            className="home-section-marketing-btn"
          >
            Drive Now
          </div>
        </div>
      </div>
      <div
        className="home-section home-second"
      >
        <div
          className="home-decorations home-decorations-horizontal"
        >
          <div
            className="home-decorations-item"
          />
          <div
            className="home-decorations-item"
          />
        </div>
        <div
          className="home-decorations home-decorations-horizontal"
        >
          <div
            className="home-decorations-item"
          />
          <div
            className="home-decorations-item"
          />
        </div>
        <div
          className="home-second-main"
          data-aos="fade-right"
        >
          <div
            className="home-second-main-body"
          >
            <p
              className="home-second-main-body-pre"
            >
              BEST SERVICES
            </p>
            <p
              className="home-second-main-body-header"
            >
              Feel the best experience with our selection deals
            </p>
            <div
              className="home-second-main-body-section"
            >
              <p
                className="home-second-main-body-section-header"
              >
                Packages for every budget
              </p>
              <p
                className="home-second-main-body-section-body"
              >
                Feel the best experience with our selection deals
              </p>
            </div>
            <div
              className="home-second-main-body-section"
            >
              <p>
                Best price
              </p>
              <p>
                We find the best deals for their price
              </p>
            </div>
            <div
              className="home-second-main-body-section"
            >
              <p>
                Support 24/7
              </p>
              <p>
                Feel free to contact our helpline 24/7
              </p>
            </div>
          </div>
          <div
            className="home-second-main-image"
            data-aos="fade-left"
          >
            <img
              src="aston_martin_PNG91.png"
            />
          </div>
        </div>
      </div>
      <div
        className="home-section home-third"
        data-aos="fade-up"
      >
        <div
          className="home-third-header"
        >
          Reviews
        </div>
        <div
          className="home-third-body"
        >
          <div
            className="home-third-body-reviews"
          >
            <div
              className="home-third-body-reviews-item"
            >
              <div
                className="home-third-body-review-item-icon"
              >
                <img
                  src="icon_user.png"
                />
              </div>
              <div
                className="home-third-body-review-item-userName"
              >
                Kiril Johnson
              </div>
              <div
                className="home-third-body-review-item-review"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised
              </div>
            </div>
            <div
              className="home-third-body-reviews-item"
            >
              <div
                className="home-third-body-review-item-icon"
              >
                <img
                  src="icon_user.png"
                />
              </div>
              <div
                className="home-third-body-review-item-userName"
              >
                Kiril Johnson
              </div>
              <div
                className="home-third-body-review-item-review"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised
              </div>
            </div>
            <div
              className="home-third-body-reviews-item"
            >
              <div
                className="home-third-body-review-item-icon"
              >
                <img
                  src="icon_user.png"
                />
              </div>
              <div
                className="home-third-body-review-item-userName"
              >
                Kiril Johnson
              </div>
              <div
                className="home-third-body-review-item-review"
              >
                Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="footerWrapper"
      >
        <h5>
          Â© 2022 CARSA, Inc.
        </h5>
      </div>
    </div>
  </div>,
  " // Now App has access to context",
]
`;

exports[`components should render correct home page test 1`] = `
Array [
  " // Set context",
  <div
    className="home"
  >
    <div
      className="home-decorations"
    >
      <div
        className="home-decorations-item"
      />
      <div
        className="home-decorations-item"
      />
    </div>
    <div
      className="home-decorations"
    >
      <div
        className="home-decorations-item"
      />
      <div
        className="home-decorations-item"
      />
    </div>
    <div
      className="homeheader"
    >
      <div
        className="homeheader-wrapper"
      >
        <a
          href="/home"
          onClick={[Function]}
        >
          Home
        </a>
      </div>
      <div
        className="homeheader-wrapper"
      >
        <a
          href="/offers"
          onClick={[Function]}
        >
          Offers
        </a>
      </div>
      
      <div
        className="homeheader-logo"
      >
        <img
          src="logo.png"
        />
      </div>
      
      <div
        className="homeheader-wrapper"
      >
        <a
          href="/about"
          onClick={[Function]}
        >
          Contact
        </a>
      </div>
      <div
        className="homeheader-wrapper"
      >
        <a
          className="header-wrapper-link"
          href="/login"
          onClick={[Function]}
        >
          Log in
        </a>
      </div>
    </div>
    <div
      className="home-section"
    >
      <div
        className="home-section-first"
      >
        <div
          className="home-section-first-fadingText"
          data-aos="fade-up"
        >
          CARSA
        </div>
        <img
          data-aos="fade-down"
          src="brandCar.png"
        />
      </div>
      <div
        className="home-section-marketing"
      >
        <p>
          Selection helper
        </p>
        <img
          src="Arrow 2.png"
        />
        <div
          className="home-section-marketing-btn"
        >
          Drive Now
        </div>
      </div>
    </div>
    <div
      className="home-section home-second"
    >
      <div
        className="home-decorations home-decorations-horizontal"
      >
        <div
          className="home-decorations-item"
        />
        <div
          className="home-decorations-item"
        />
      </div>
      <div
        className="home-decorations home-decorations-horizontal"
      >
        <div
          className="home-decorations-item"
        />
        <div
          className="home-decorations-item"
        />
      </div>
      <div
        className="home-second-main"
        data-aos="fade-right"
      >
        <div
          className="home-second-main-body"
        >
          <p
            className="home-second-main-body-pre"
          >
            BEST SERVICES
          </p>
          <p
            className="home-second-main-body-header"
          >
            Feel the best experience with our selection deals
          </p>
          <div
            className="home-second-main-body-section"
          >
            <p
              className="home-second-main-body-section-header"
            >
              Packages for every budget
            </p>
            <p
              className="home-second-main-body-section-body"
            >
              Feel the best experience with our selection deals
            </p>
          </div>
          <div
            className="home-second-main-body-section"
          >
            <p>
              Best price
            </p>
            <p>
              We find the best deals for their price
            </p>
          </div>
          <div
            className="home-second-main-body-section"
          >
            <p>
              Support 24/7
            </p>
            <p>
              Feel free to contact our helpline 24/7
            </p>
          </div>
        </div>
        <div
          className="home-second-main-image"
          data-aos="fade-left"
        >
          <img
            src="aston_martin_PNG91.png"
          />
        </div>
      </div>
    </div>
    <div
      className="home-section home-third"
      data-aos="fade-up"
    >
      <div
        className="home-third-header"
      >
        Reviews
      </div>
      <div
        className="home-third-body"
      >
        <div
          className="home-third-body-reviews"
        >
          <div
            className="home-third-body-reviews-item"
          >
            <div
              className="home-third-body-review-item-icon"
            >
              <img
                src="icon_user.png"
              />
            </div>
            <div
              className="home-third-body-review-item-userName"
            >
              Kiril Johnson
            </div>
            <div
              className="home-third-body-review-item-review"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised
            </div>
          </div>
          <div
            className="home-third-body-reviews-item"
          >
            <div
              className="home-third-body-review-item-icon"
            >
              <img
                src="icon_user.png"
              />
            </div>
            <div
              className="home-third-body-review-item-userName"
            >
              Kiril Johnson
            </div>
            <div
              className="home-third-body-review-item-review"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised
            </div>
          </div>
          <div
            className="home-third-body-reviews-item"
          >
            <div
              className="home-third-body-review-item-icon"
            >
              <img
                src="icon_user.png"
              />
            </div>
            <div
              className="home-third-body-review-item-userName"
            >
              Kiril Johnson
            </div>
            <div
              className="home-third-body-review-item-review"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="footerWrapper"
    >
      <h5>
        Â© 2022 CARSA, Inc.
      </h5>
    </div>
  </div>,
  " // Now App has access to context",
]
`;

exports[`components should render correct login page test 1`] = `
Array [
  " // Set context",
  <div
    className="wrapper"
  >
    <div
      className="header-wrapper"
    >
      <a
        className="header-wrapper-link"
        href="/home"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="header-wrapper-link"
        href="/offers"
        onClick={[Function]}
      >
        Offers
      </a>
      <a
        className="header-wrapper-link"
        href="/about"
        onClick={[Function]}
      >
        About Us
      </a>
      <a
        className="header-wrapper-link"
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
      
      <a
        className="header-wrapper-link"
        href="/login"
        onClick={[Function]}
      >
        Log in
      </a>
    </div>
    <div
      className="wrapper"
    >
      <h1
        className="formLabel"
      >
        Log In
      </h1>
      <form
        action="#"
        noValidate={true}
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="wrapper"
        >
          <input
            className=""
            id="email"
            name="email"
            placeholder="EMAIL"
            type="email"
            value=""
          />
        </div>
        <div
          className="wrapper"
        >
          <input
            className=""
            id="password"
            name="password"
            placeholder="PASSWORD"
            type="password"
            value=""
          />
        </div>
        <a
          className="forgotPasswordLink"
          href="/home"
          onClick={[Function]}
        >
          Forgot password?
        </a>
        <button
          className="submitButton"
          type="submit"
        >
          Log in
        </button>
      </form>
      <a
        className="signupLink"
        href="/register"
        onClick={[Function]}
      >
        Sign up now
      </a>
    </div>
  </div>,
  " // Now App has access to context",
]
`;
